packages/hris-kernel/tests/test_assignment_portfolio.py

```
@@ -67,6 +67,26 @@ def test_portfolio_rejects_allocation_above_one_for_one_employment(
 
 
+def test_portfolio_rejects_exact_overallocation_under_low_decimal_precision(
+    jordan_icu_assignment,
+    jordan_float_assignment,
+) -> None:
+    """Allocation limits must use exact totals even when Decimal precision is low."""
+    first = replace(jordan_icu_assignment, allocation_ratio=Decimal("0.5040"))
+    second = replace(jordan_float_assignment, allocation_ratio=Decimal("0.5040"))
+    with localcontext() as context:
+        context.prec = 2
+        with pytest.raises(AssignmentPortfolioError, match="1.0000"):
```

Ruff (0.16.2)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)